cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/FileBatchSource.java

```diff
@@ -158,9 +162,18 @@ public void prepareRun(BatchSourceContext context) throws Exception {
       Class<? extends FileInputFormat> classType = (Class<? extends FileInputFormat>)
         classLoader.loadClass(config.inputFormatClass);
       job.setInputFormatClass(classType);
+    } else {
+      job.setInputFormatClass(CombineTextInputFormat.class);
     }
     FileInputFormat.setInputPathFilter(job, BatchFileFilter.class);
     FileInputFormat.addInputPath(job, new Path(config.path));
+    long maxSplit;
+    try {
+      maxSplit = Long.valueOf(config.maxSplitSize);
```

Comment:
``maxSplit`` is a primitive ``long``. So you don't need ``valueOf``, which returns a ``new Long()``. Use ``Long.parseLong`` instead.